<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>OutboundEditor</class>
 <widget class="QDialog" name="OutboundEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>596</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>920</width>
    <height>570</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Edit Connection Settings</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_4" columnstretch="2,4">
   <item row="1" column="1">
    <widget class="QGroupBox" name="streamSettingsGroupBox">
     <property name="title">
      <string>Stream Settings</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <layout class="QGridLayout" name="transportFrame"/>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QFormLayout" name="formLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="ipLabel_3">
         <property name="text">
          <string>Tag</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="tagTxt">
         <property name="placeholderText">
          <string>Tag of this outbound setting</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_25">
         <property name="text">
          <string>Type</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="outBoundTypeCombo">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <item>
          <property name="text">
           <string notr="true">VMess</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">ShadowSocks</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Socks</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,5,0,1">
         <item>
          <widget class="QLabel" name="ipLabel">
           <property name="text">
            <string>Host</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="ipLineEdit">
           <property name="placeholderText">
            <string>Hostname or IP/IPv6 Address</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="portLabel">
           <property name="text">
            <string>Port</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="portLineEdit">
           <property name="maxLength">
            <number>5</number>
           </property>
           <property name="placeholderText">
            <string>Port</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Misc Settings</string>
       </property>
       <layout class="QFormLayout" name="formLayout_8">
        <item row="0" column="0">
         <widget class="QLabel" name="label_32">
          <property name="text">
           <string>Use Mux</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="muxEnabledCB">
          <property name="text">
           <string>Enabled</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_33">
          <property name="text">
           <string>Mux Concurrency</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="muxConcurrencyTxt">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1024</number>
          </property>
          <property name="value">
           <number>8</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="useFPCB">
          <property name="text">
           <string>Enabled</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_23">
          <property name="text">
           <string>Use Forward Proxy</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="outboundSettingWrapper">
     <property name="title">
      <string>Outbound Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="3">
        <item>
         <widget class="QStackedWidget" name="outboundTypeStackView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page_2">
           <layout class="QFormLayout" name="formLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="idLabel">
              <property name="text">
               <string>UUID</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="idLineEdit">
              <property name="text">
               <string/>
              </property>
              <property name="maxLength">
               <number>36</number>
              </property>
              <property name="placeholderText">
               <string notr="true">xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="alterLabel">
              <property name="text">
               <string>Alter ID</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="alterLineEdit">
              <property name="maximum">
               <number>1024</number>
              </property>
              <property name="value">
               <number>16</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="securityLabel">
              <property name="text">
               <string>Security</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="securityCombo">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <item>
               <property name="text">
                <string notr="true">auto</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">aes-128-gcm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">chacha20-poly1305</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">none</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page">
           <layout class="QFormLayout" name="formLayout_6">
            <item row="0" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Email</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_22">
              <property name="text">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Encryption Method</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="ss_encryptionMethod">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <item>
               <property name="text">
                <string notr="true">aes-256-cfb</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">aes-128-cfb</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">aes-256-gcm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">aes-128-gcm</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">chacha20-ietf</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">chacha20</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">chacha20-poly1305</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">chacha20-ietf-poly1305</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_24">
              <property name="text">
               <string>Level</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="ss_levelSpin"/>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_30">
              <property name="text">
               <string>OTA</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QCheckBox" name="ss_otaCheckBox">
              <property name="text">
               <string>Enabled</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="ss_emailTxt">
              <property name="placeholderText">
               <string notr="true">my.name@domain.com</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="ss_passwordTxt">
              <property name="placeholderText">
               <string notr="true">p@ssw0rd</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_3">
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_21">
              <property name="text">
               <string>Username</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="socks_UserNameTxt">
              <property name="placeholderText">
               <string notr="true">Username</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_31">
              <property name="text">
               <string>Password</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="socks_PasswordTxt">
              <property name="placeholderText">
               <string notr="true">p@ssw0rd</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tagTxt</tabstop>
  <tabstop>ipLineEdit</tabstop>
  <tabstop>portLineEdit</tabstop>
  <tabstop>outBoundTypeCombo</tabstop>
  <tabstop>idLineEdit</tabstop>
  <tabstop>alterLineEdit</tabstop>
  <tabstop>securityCombo</tabstop>
  <tabstop>ss_emailTxt</tabstop>
  <tabstop>ss_passwordTxt</tabstop>
  <tabstop>ss_encryptionMethod</tabstop>
  <tabstop>ss_levelSpin</tabstop>
  <tabstop>ss_otaCheckBox</tabstop>
  <tabstop>socks_UserNameTxt</tabstop>
  <tabstop>socks_PasswordTxt</tabstop>
  <tabstop>muxEnabledCB</tabstop>
  <tabstop>muxConcurrencyTxt</tabstop>
  <tabstop>useFPCB</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OutboundEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>601</x>
     <y>586</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OutboundEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>669</x>
     <y>586</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
